### PR TITLE
Using cluster_seeds means incremental clustering

### PR DESCRIPTION
**Pull request by stefanc-ai2**

### CODE REVIEW

File: s2apler/model.py [hunk 1]
```diff
@@ -512,7 +512,6 @@ def predict(
                 else:
                     block_dict_no_null_titles[block_key].append(paper_id)
         block_dict = dict(block_dict_no_null_titles)
-        block_dict_no_null_titles = dict(block_dict_null_titles)
```
Comment (stefanc-ai2) [inline s2apler/model.py:515]: Removing this swapped variable naming to prevent confusion

File: s2apler/model.py [hunk 2]
```diff
@@ -569,7 +568,7 @@ def predict(
                 max_label = 0
 
             # now we can rejoin the null titles based only on the ids
-            paper_ids_for_nulls = block_dict_no_null_titles.get(block_key, [])
+            paper_ids_for_nulls = block_dict_null_titles.get(block_key, [])
```
Comment (sergeyf) [inline s2apler/model.py:571]: Is this a bug? What's going on with the changes in model.py?

File: s2apler/timo/interface.py [hunk 3]
```diff
@@ -141,7 +140,7 @@ def predict_one(self, instance: Instance) -> Prediction:
             mode="inference",
             n_jobs=self._config.n_jobs,
         )
-        if self._config.predict_incremental:
+        if instance.cluster_seeds:
```
Comment (sergeyf) [inline s2apler/timo/interface.py:147]: Why not use the config? It's more explicit...

File: s2apler/model.py (hunk 2 above)
Comment (stefanc-ai2) [reply s2apler/model.py:571]: Not a bug, but the variable naming above is confusing or swapped.

File: s2apler/model.py (hunk 1 above)
Comment (sergeyf) [reply s2apler/model.py:515]: It also appears in rows 513 and 514. Won't those rows error out?

Comment (sergeyf) [reply s2apler/model.py:515]: Nevermind, I see what's happening. Thanks.

File: s2apler/timo/interface.py (hunk 3 above)
Comment (stefanc-ai2) [reply s2apler/timo/interface.py:147]: If we use the config, we'll need to have separate endpoint/deployment for incremental mode. It'll be nice for the clustering worker to send requests to the same endpoint. Also, this removes the possibility of user thinking they can send `cluster_seeds` in full block mode.

Comment (sergeyf) [reply s2apler/timo/interface.py:147]: OK, but can we add some kind of error message to catch this situation. How will a user know they can't send `cluster_seeds` in full block mode?

Comment (stefanc-ai2) [reply s2apler/timo/interface.py:147]: The config is tied to the whole endpoint/deployment. It's not a parameter that can be specified on a per request basis.

Comment (stefanc-ai2) [reply s2apler/timo/interface.py:147]: > OK, but can we add some kind of error message to catch this situation. How will a user know they can't send `cluster_seeds` in full block mode?

Yes, sure thing.

Comment (sergeyf) [reply s2apler/timo/interface.py:147]: I see. We probably just need to document this more extensively here in the comments then.

**Review by sergeyf — APPROVED** [review]
LGTM

**Pull request merged**